R/plot.data_table_threads_benchmark.R

```diff
@@ -28,26 +28,26 @@ plot.data_table_threads_benchmark <- function(x, ...)
     type = "Measured"
   ), by = expr]
 
-  maxSpeedup <- x[, .(threadCount = threadCount[which.max(speedup)],
+maxSpeedup <- x[, .(threadCount = threadCount[which.max(speedup)],
                       speedup = max(speedup),
-                      minTime = as.numeric(NA),
-                      maxTime = as.numeric(NA),
-                      medianTime = as.numeric(NA),
+                      minTime = NA_real_,
+                      maxTime = NA_real_,
```

Comment:
It's because of non-existing columns when combining the data.tables and setting the order using `setcolorder` (`colnamesInt` would throw an error otherwise), more specifically `minTime`, `maxTime`, and `medianTime` don't exist in those tables (`idealSpeedup`, `recommendedSpeedup`, `maxSpeedup`, and `closestPoints`) so I introduced those initializations as a quick fix, but you're right that this is not a good approach. I was thinking to remove the redundancy or repetition by having a loop:
```r
for (dt in list(x, idealSpeedup, recommendedSpeedup, maxSpeedup)) {
  for (col in setdiff(cols, names(dt))) {
    dt[, (col) := NA_real_]
  }
}
```

But since you mentioned initializing to NA is bad/unusual, I figured that I could just implement a dynamic column order adjustment using `setcolorder` for each `data.table` instead. (Let me know what you think about this change!)

Reply:
if those columns do not exist then probably you should not be trying to combine those tables

Reply:
My bad, they did exist but the column order was messed up

Reply:
I refactored my code since then and I think I got all your points, so could you please do a final review now?